services/net/clip/ClipAction.cs

```diff
@@ -56,55 +56,66 @@ public ClipAction(IKafkaMessenger kafka, IApiService api, IOptions<ClipOptions>
+                    }
+                    else if (reference.WorkflowStatus == (int)WorkflowStatus.InProgress && reference.UpdatedOn?.AddMinutes(2) < DateTime.UtcNow)
+                    {
+                        // If another process has it in progress only attempt to do an import if it's
+                        // more than an 2 minutes old. Assumption is that it is stuck.
+                        reference = await this.Api.UpdateContentReferenceAsync(reference);
+                    }
+                    else sendMessage = false;
+
+                    if (sendMessage && reference != null)
+                    {
+                        // TODO: Waiting for each clip to complete isn't ideal.  It needs to handle multiple processes.
+                        await RunProcessAsync(process, cancellationToken);
+                        var messageResult = await SendMessageAsync(manager.DataSource, schedule, reference);
+
+                        reference.WorkflowStatus = (int)WorkflowStatus.Received;
+                        reference.Partition = messageResult.Partition;
+                        reference.Offset = messageResult.Offset;
+                        await this.Api.UpdateContentReferenceAsync(reference);
+                    }
                 }
-                else if (reference.WorkflowStatus == (int)WorkflowStatus.InProgress && reference.UpdatedOn?.AddMinutes(2) < DateTime.UtcNow)
+                else if (name == "stop")
                 {
-                    // If another process has it in progress only attempt to do an import if it's
-                    // more than an 2 minutes old. Assumption is that it is stuck.
-                    reference = await this.Api.UpdateContentReferenceAsync(reference);
-                }
-                else sendMessage = false;
-
-                if (sendMessage && reference != null)
-                {
-                    // TODO: Waiting for each clip to complete isn't ideal.  It needs to handle multiple processes.
-                    await RunProcessAsync(process, cancellationToken);
-                    var messageResult = await SendMessageAsync(manager.DataSource, schedule, reference);
-
-                    reference.WorkflowStatus = (int)WorkflowStatus.Received;
-                    reference.Partition = messageResult.Partition;
-                    reference.Offset = messageResult.Offset;
-                    await this.Api.UpdateContentReferenceAsync(reference);
+                    await StopProcessAsync(process, cancellationToken);
+                    RemoveProcess(manager, schedule);
                 }
             }
-            else if (name == "stop")
+            catch (Exception ex)
             {
-                await StopProcessAsync(process, cancellationToken);
-                RemoveProcess(manager, schedule);
+                if ((ex is MissingFileException || (ex is AggregateException && ex.InnerException is MissingFileException)) &&
```

Comment:
This is the new code to fix the bug.  The rest is the result of wrapping in the `try+catch`